test/integration/api_keys_test.rb

```diff
@@ -218,10 +218,13 @@ class ApiKeysTest < SystemTest
 
     visit_profile_api_keys_path
     assert page.has_css? ".owners__row__invalid"
+    assert api_key.reload.invalid?
+    # looks a bit weird bc there's no gem metadata available
     assert page.has_content? "(ownership removed)"
 
+    # Check if edit button has disappeared
     click_button "Edit"
-    assert page.has_css? ".flash"
+    # Update behaviour to redirect back to api keys page if gone to edit path
```

Comment:
Next steps: make this into a reality

Reply:
https://github.com/Shopify/rubygems.org/pull/14/commits/b917177bea23883bc0dd05d950ab6cdb741369a1